### PR TITLE
Use schema library in test scope only

### PR DESCRIPTION
**Pull request by timja**
Seems like this dep isn't used in production code, should hopefully fix: https://github.com/jenkinsci/bom/pull/111

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: plugin/pom.xml [hunk 1]
```diff
@@ -50,6 +50,18 @@
       <groupId>org.everit.json</groupId>
       <artifactId>org.everit.json.schema</artifactId>
       <version>1.5.1</version>
+      <scope>test</scope>
```
Comment (sladyn98) [inline plugin/pom.xml]: Adding this to the test scope sounds logical to me, I am not quite sure what `commons-beanutils` does.

**Comment by casz** [comment]
Do we really need both GSON and json.org?

**Comment by timja** [comment]
> Do we really need both GSON and json.org?
not really related to the PR, this just fixes the existing packaging 😹

@sladyn98 any reason you're using both gson and json.org?

**Comment by timja** [comment]
> Ya, seems like a we do not.
> 
> Looking at the imports
> https://github.com/jenkinsci/configuration-as-code-plugin/blob/73f59591c21549a245c5fbc71e3f006b553c2e6d/plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java#L3-L14
> 
> we can use GSON's [JsonObject](https://static.javadoc.io/com.google.code.gson/gson/2.6.2/com/google/gson/JsonObject.html) and [JsonArray](https://static.javadoc.io/com.google.code.gson/gson/2.6.2/com/google/gson/JsonArray.html)

I would raise a separate issue for that...

**Comment by casz** [comment]
Not really since your adding JSON.org back 😆 it was there transitively before but still...

**Comment by timja** [comment]
> Not really since your adding JSON.org back 😆 it was there transitively before but still...

It was used in production code, but only relied on transitively, adding the dependency explicitly doesn't change the code.

**Comment by timja** [comment]
@casz SchemaLoader takes json.org as an input
https://github.com/everit-org/json-schema/blob/master/core/src/main/java/org/everit/json/schema/loader/SchemaLoader.java#L282

**Comment by casz** [comment]
There is a helper method for that use case:
https://github.com/everit-org/json-schema/blob/721c1bcdec0df5dffc574a8cbf42099cc74c11ee/core/src/main/java/org/everit/json/schema/loader/SchemaLoader.java#L41-L43

**Comment by timja** [comment]
I’ve changed some of the code locally to use gson, I don’t really like the api.

It makes the code a lot harder to read and more verbose json.org has a nice fluent api, while gson rejected it
https://github.com/google/gson/issues/1031

**Comment by casz** [comment]
I am all for removing GSON :)

**Comment by sladyn98** [comment]
`String prettyJsonString = gson.toJson(jsonElement);`
I guess its used to prettyPrint the json. I am not quite sure if there is an alternative , do let me know

**Comment by casz** [comment]
@sladyn98 there is https://stackoverflow.com/a/9583835/1103925

```java
JSONObject json = new JSONObject(jsonString); // Convert text to object
System.out.println(json.toString(4)); // Print it with specified indentation
```

**Comment by casz** [comment]
A upstream PR to BOM would be a good idea 💡

**Comment by sladyn98** [comment]
@timja Looks like we will have to make sure that the new Testing PR reflects these changes or else it would just appear again :)

**Comment by timja** [comment]
CI is dead, merging now so less re-work in @sladyn98 's PRs later

**Pull request merged**